test/Rector/Class_/ImplementsFactoryInterfaceToPsrFactoryRector/FixtureAutoImportRenameUse/implements_root_factory_interface_call_other_type_hint_method.php.inc

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\ServiceManager\Migration\Rector\Class_\FixtureAutoImportRenameUse\Fixture;
+
+use Interop\Container\ContainerInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+
+class ImplementsRootAbstractFactoryInterfaceCallOtherTypeHintMethod implements FactoryInterface
+{
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
+    {
+        $this->verify($container);
+    }
+    private function verify(ContainerInterface $container): void
+    {
+    }
+}
+
+?>
+-----
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\ServiceManager\Migration\Rector\Class_\FixtureAutoImportRenameUse\Fixture;
+
+use Laminas\ServiceManager\Factory\FactoryInterface;
+
+class ImplementsRootAbstractFactoryInterfaceCallOtherTypeHintMethod
+{
+    public function __invoke(\Psr\Container\ContainerInterface $container)
+    {
+        $this->verify($container);
+    }
+    private function verify(\Psr\Container\ContainerInterface $container): void
+    {
+    }
```

Comment:
@froschdesign it solved now in rector 0.12.7, when auto import enabled, it will auto apply to use statements even type hint used in other methods https://github.com/laminas/laminas-servicemanager-migration/pull/35/commits/610a1b6a9cdf58ae484dde9d5f70af265719bee9